wasm/webapi/wasm_stream_compile_test.html

```diff
@@ -1,125 +1,115 @@
   };
 
   promise_test(async function() {
-      var result = fetch(getWasmUrl('resources/incrementer.wasm'), {"mode": "cors"} );
+      const result = fetch(getWasmUrl('resources/incrementer.wasm'), {"mode": "cors"} );
       const module = await WebAssembly.compileStreaming(result);
       assert_true(module instanceof WebAssembly.Module);
   }, "compileStreaming check CORS");
 
-  promise_test(async function() {
-    try {
-      var result = fetch(getWasmUrl('resources/incrementer.wasm'), {"mode": "no-cors"} );
-      const module = await WebAssembly.compileStreaming(result);
-      assert_true(false);
-    } catch (e) {
-      assert_equals(e.message, "Response is not CORS-same-origin");
-      assert_true(e instanceof Error);
-    }
+  promise_test(async function(t) {
+      const result = fetch(getWasmUrl('resources/incrementer.wasm'), {"mode": "no-cors"} );
+      await promise_rejects_js(t, TypeError, WebAssembly.compileStreaming(result));
   }, "compileStreaming raise error if no-cors");
 
   promise_test(async function() {
-    const module = await fetch('resources/incrementer.wasm')
-      .then(v => v.arrayBuffer())
-      .then(v => WebAssembly.compileStreaming(new Response(v, { headers: { "Content-Type" : "application/wasm" }})));
-    assert_true(module instanceof WebAssembly.Module);
+      const v = await fetch('resources/incrementer.wasm');
+      const buffer = await v.arrayBuffer();
+      const response = new Response(buffer, { headers: { "Content-Type" : "application/wasm" }});
+      const module = await WebAssembly.compileStreaming(response);
+      assert_true(module instanceof WebAssembly.Module);
   }, "compileStreaming receive promise with response created from ArrayBuffer");
 
   promise_test(async function() {
-    const module = await fetch('resources/incrementer.wasm')
-      .then(v => v.arrayBuffer())
-      .then(buffer => {
-        const stream = new ReadableStream({ 
-          start(controller) {
-            (async () => {
-              await Promise.resolve().then(() => controller.enqueue(buffer.slice(0, 20)));
-              await Promise.resolve().then(() => controller.enqueue(buffer.slice(20, buffer.byteLength)));
-              await Promise.resolve().then(() => controller.close());
-            })();
-          }
-        });
-        return stream; 
-      }) 
-      .then(stream => WebAssembly.compileStreaming(new Response(stream, { headers: { "Content-Type" : "application/wasm" }})));
-    assert_true(module instanceof WebAssembly.Module);
-  }, "compileStreaming receive response that deliver data by chunks as bufferArray");
+      const v = await fetch('resources/incrementer.wasm');
+      const buffer = await v.arrayBuffer();
+      const stream = new ReadableStream({
+        start(controller) {
+          (async () => {
+            await Promise.resolve().then(() => controller.enqueue(new Uint8Array(buffer.slice(0, 20))));
```

Comment:
could this be just `await controller.enqueue(new Uint8Array(buffer.slice(0,20)))` (& likewise below)?

Reply:
That seems to work as well, but I don't know if it changes timing / the goal of the test